src/utils/http.js

```diff
@@ -0,0 +1,33 @@
+import {
+  get as rawGet,
+  post as rawPost,
+  patch as rawPatch,
+  del as rawDelete,
+} from 'axios';
+import { merge } from 'lodash/fp';
+import { LOGIN_URL } from '../config/routing';
+import { LOGIN_ENDPOINT } from '../config/endpoints';
+
+const handleUnAuthorizedError = (error) => {
+  const { response: { status, config: { url } } } = error;
+  if (status === 401 && url !== LOGIN_ENDPOINT) window.location = LOGIN_URL;
+  throw error;
+};
+
+const createHeaderWithBearerToken = (token) => ({
+  headers: {
+    Authorization: `Bearer ${token}`
+  }
+});
+
+export const sendWithUserAuthToken = (httpMethod) => (url, config = {}, token = '') => {
+  const authHeader = createHeaderWithBearerToken(token);
+  return httpMethod(url, merge(config, authHeader))
+    .then((res) => res.data)
+    .catch(handleUnAuthorizedError);
```

Comment:
Will it be hard to catch more than one type of error?

Reply:
We can simply handle it in this catch or add another catch if we need.